internal/provider/metric_export_prometheus_config_resource.go

```diff
@@ -0,0 +1,405 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/cockroachdb/cockroach-cloud-sdk-go/pkg/client"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"net/http"
+	"strings"
+)
+
+var metricExportPrometheusConfigAttributes = map[string]schema.Attribute{
+	"id": schema.StringAttribute{
+		Required:            true,
+		MarkdownDescription: "Cluster ID.",
+		PlanModifiers: []planmodifier.String{
+			stringplanmodifier.RequiresReplace(),
+		},
+	},
+	"enabled": schema.BoolAttribute{
+		Default:     booldefault.StaticBool(true),
+		Optional:    true,
+		Computed:    true,
+		Description: "Represents prometheus integration status. Please refrain setting any value as it is used for state management.",
+	},
+	"status": schema.StringAttribute{
+		Computed:    true,
+		Description: "Encodes the possible states that a metric export configuration can be in as it is created, deployed, and disabled.",
+	},
+	"user_message": schema.StringAttribute{
```

Comment:
I don't really love the name `user_message` for this.  I don't think we need to change the api but we could choose a better name for the terraform resource field.  Maybe `status_detail` would be a better name here. @abarganier can you help with this one too?